packages/zwave-js/src/lib/controller/Controller.ts

```diff
@@ -898,6 +904,64 @@ export class ZWaveController extends EventEmitter {
 		return true; // Don't invoke any more handlers
 	}
 
+	/**
+	 * Is called when an ReplaceFailed request is received from the controller.
+	 * Handles and controls the replace process.
+	 */
+	private async handleReplaceNodeRequest(
+		msg: ReplaceFailedNodeRequest,
+	): Promise<boolean> {
+		log.controller.print(
+			`handling replace node request (status = ${
+				ReplaceFailedNodeStatus[msg.status!]
+			})`,
+		);
+
+		switch (msg.status) {
+			case ReplaceFailedNodeStatus.NodeOK:
+				if (this._replaceFailedPromise != null) {
+					this._replaceFailedPromise.reject(
+						new ZWaveError(
+							`The node ${msg.failedNodeId} could not be replaced because it's not failed`,
+							ZWaveErrorCodes.ReplaceFailedNode_NodeOK,
+						),
+					);
+				}
+				break;
+			case ReplaceFailedNodeStatus.FailedNodeReplaceFailed:
+				if (this._replaceFailedPromise != null) {
+					this._replaceFailedPromise.reject(
+						new ZWaveError(
+							`The replace process could not be completed`,
+							ZWaveErrorCodes.ReplaceFailedNode_Failed,
+						),
+					);
+				}
+				break;
+			case ReplaceFailedNodeStatus.FailedNodeReplace:
+				// failed node is now ready to be replaced and controller is ready to add a new
+				// node with the nodeID of the failed node
+				await this.beginInclusion(msg.includeNonSecure);
```

Comment:
`_includeNonSecure` is a local variable of the controller, not a property of the received message. You should set it when starting the replacement process.

Reply:
Humm msg is instance of ReplaceFailedNodeRequest, It's a property of the request if you check that, like `failedNodeId`

Reply:
> You should set it when starting the replacement process

It is handled when I call `beginInlcusion`

Reply:
The argument (msg) here is the message that is **received** from the controller (the ReplaceFailedNodeRequestStatusReport, `ZW->HOST: REQ | 0x63 | funcID | txStatus` in the specs). 
It won't have the property set.

Reply:
Fixed